substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/reports/CallTreePrinter.java

```diff
@@ -300,4 +314,268 @@ private static String packagePrefix(String name) {
         }
         return name.substring(0, lastDot);
     }
+
+    private static void printCsvFiles(Map<AnalysisMethod, MethodNode> methodToNode, String path, String reportName) {
+        // Set virtual node at next available method id
+        final AtomicInteger virtualNodeId = new AtomicInteger(MethodNode.methodId);
+
+        Set<Integer> entryPointIds = new HashSet<>();
+        Set<MethodNode> nonVirtualNodes = new HashSet<>();
+        Map<List<String>, Integer> virtualNodes = new HashMap<>();
+
+        Map<Integer, Set<BciEndEdge>> directEdges = new HashMap<>();
+        Map<Integer, Set<BciEndEdge>> virtualEdges = new HashMap<>();
+        Map<Integer, Set<Integer>> overridenByEdges = new HashMap<>();
+
+        final Iterator<MethodNode> iterator = methodToNode.values().stream().filter(n -> n.isEntryPoint).iterator();
+        while (iterator.hasNext()) {
+            final MethodNode node = iterator.next();
+            entryPointIds.add(node.id);
+            walkNodes(node, directEdges, virtualEdges, overridenByEdges, virtualNodes, nonVirtualNodes, virtualNodeId);
+        }
+
+        ReportUtils.report("call tree for vm entry point", path + File.separatorChar + "reports", "csv_call_tree_vm_" + reportName, "csv",
+                        CallTreePrinter::printVMEntryPoint);
```

Comment:
If I understand correctly you need all the .csv files to inspect the graphs. In that case using this `report` method is not ideal as it will timestamp each file (as @Sanne also noticed). You could instead use `com.oracle.graal.pointsto.reports.ReportUtils#report(java.lang.String, java.nio.file.Path, java.util.function.Consumer<java.io.PrintWriter>)` and provide a stable name for each file. To keep files organized you can write all .csv files for a run into a directory and timestamp the directory.

Reply:
I was trying to follow convention with the timestamp, but indeed as @Sanne noted, it makes scripting complicated. Timestamp the directory and you have the same issue. I think the `report` folder is just fine as is, but we can use symlinks to link to the latest csv files, e.g. `report/csv_call_tree_vm.csv` links to `report/csv_call_tree_vm_123457890`. Whenever a new execution happens, the symlink is updated to the latest one. This would help with scripting providing stable paths.